### PR TITLE
Add methods to wallet for batch processing

### PR DESCRIPTION
**Pull request by lalexgap**
This PR introduces three new methods to the wallet API:
- JoinChannels - Allows the joining of multiple channels in one call to the wallet.
- UpdateFundingForChannels - Allows the updating of funding for multiple channels in one call to the wallet.
- MergeMessages - Merges multiple messages into one message. This is useful for merging the result of different API calls in one request/response cycle.

### CODE REVIEW

File: packages/server-wallet/src/wallet/index.ts [hunk 1]
```diff
@@ -66,14 +71,15 @@ export type WalletInterface = {
...
   updateChannel(args: UpdateChannelParams): SingleChannelResult;
   closeChannel(args: CloseChannelParams): SingleChannelResult;
   getChannels(): MultipleChannelResult;
   getState(args: GetStateParams): SingleChannelResult;
   syncChannel(args: SyncChannelParams): SingleChannelResult;
 
-  updateChannelFunding(args: UpdateChannelFundingParams): void;
-
+  updateChannelFunding(args: UpdateChannelFundingParams): SingleChannelResult;
+  updateFundingForChannels(args: UpdateChannelFundingParams[]): MultipleChannelResult;
```
Comment (kerzhner) [inline packages/server-wallet/src/wallet/index.ts:82]: Could be nice for the naming to be consistent. So maybe `updateChannelFunding`/`updateChannelsFunding` or `updateFundingForChannel`/`updateFundingForChannels`

Comment (kerzhner) [reply packages/server-wallet/src/wallet/index.ts:82]: Another option is to have just one api call `updateChannelsFunding` that works for 1 or more channels.

File: packages/server-wallet/src/wallet/index.ts [hunk 2]
```diff
@@ -240,6 +273,19 @@ export class Wallet implements WalletInterface {
     );
     return {outbox: mergeOutgoing(outgoing.map(n => n.notice)), channelResult};
   }
+
+  async joinChannels(channelIds: ChannelId[]): MultipleChannelResult {
```
Comment (kerzhner) [inline packages/server-wallet/src/wallet/index.ts:277]: Do we need `joinChannel` if we have a `joinChannels` that already works for a single element `channelIds` array?

File: packages/server-wallet/src/wallet/index.ts [hunk 3]
```diff
@@ -102,11 +110,13 @@ export class Wallet implements WalletInterface {
     // Bind methods to class instance
     this.getParticipant = this.getParticipant.bind(this);
```
Comment (kerzhner) [inline packages/server-wallet/src/wallet/index.ts:111]: Probably doesn't belong in this PR, but it might be time to refactor this code to:
```
const methods = [this.getParticipants, this.updateChannelFunding, ...];
methods.map(m => m.bind(this))
```

**Review by kerzhner — APPROVED** [review]

File: packages/server-wallet/src/wallet/index.ts (hunk 3 above)
Comment (andrewgordstewart) [reply packages/server-wallet/src/wallet/index.ts:111]: It would also be nice to have an explanation of why this is necessary.

File: packages/server-wallet/src/wallet/index.ts (hunk 2 above)
Comment (andrewgordstewart) [reply packages/server-wallet/src/wallet/index.ts:277]: There's a question of maintaining backwards compatibility. But I think that the server-wallet's API is closer to the One True API than the `client-api-schema`'s opinion.

Comment (lalexgap) [reply packages/server-wallet/src/wallet/index.ts:273]: I've removed `joinChannel`, `createChannel` and `updateChannelFunding` in 
1684ff4

Comment (tomclose) [reply packages/server-wallet/src/wallet/index.ts:273]: The backwards compatibility struggle is real. Especially when `createChannel` would guarantee you a `SingleChannelResult`, but `createChannels` doesn't. I suggest that we put these methods back in - it costs very little to maintain them, and we reduce the hassle burden for users of the wallet.

Comment (tomclose) [reply packages/server-wallet/src/wallet/index.ts:273]: For reference, this is [the diff](https://github.com/statechannels/the-graph/pull/180/files#diff-290c3918607a033549623003e44452b1) for the payment manager:

Screen Shot 2020-10-05 at 3 38 45 PM
